src/components/settings/SettingsSSOTab/NewSSOConfigCard.jsx

```diff
@@ -0,0 +1,192 @@
+} from '@edx/paragon';
+import PropTypes from 'prop-types';
+import {
+  Key, KeyOff, MoreVert,
+} from '@edx/paragon/icons';
+import { SSOConfigContext } from './SSOConfigContext';
+import LmsApiService from '../../../data/services/LmsApiService';
+
+const NewSSOConfigCard = ({
+  config,
+  setLoading,
+  setRefreshBool,
+  refreshBool,
+}) => {
+  const { setProviderConfig } = useContext(SSOConfigContext);
+
+  const convertToReadableDate = (date) => {
+    const dateObj = new Date(date);
+    const options = { year: 'numeric', month: 'short', day: 'numeric' };
+    return new Intl.DateTimeFormat('en-US', options).format(dateObj);
+  };
+
+  const onDeleteClick = (deletedConfig) => {
+    setLoading(true);
+    LmsApiService.deleteEnterpriseSsoOrchestrationRecord(deletedConfig.uuid).then(() => {
+      setRefreshBool(!refreshBool);
+    });
+  };
+
+  const onDisableClick = (disabledConfig) => {
+    setLoading(true);
+    LmsApiService.updateEnterpriseSsoOrchestrationRecord({ active: false }, disabledConfig.uuid).then(() => {
+      setRefreshBool(!refreshBool);
+    });
+  };
+
+  const onEnableClick = (enabledConfig) => {
+    setLoading(true);
+    LmsApiService.updateEnterpriseSsoOrchestrationRecord({ active: true }, enabledConfig.uuid).then(() => {
+      setRefreshBool(!refreshBool);
+    });
+  };
+
+  const renderCardStatusIcon = () => (
+    <>
+      {config.validated_at && config.active && (
+        <OverlayTrigger
+          id="tooltip-disabled"
+          placement="top"
+          overlay={<Tooltip variant="light">The integration is verified and working</Tooltip>}
+        >
+          <Key
+            className="bg-success-500 rounded-circle text-white mr-2 p-1"
+          />
+        </OverlayTrigger>
+      )}
+      {!config.validated_at && (
+        <OverlayTrigger
+          id="tooltip-disabled"
+          placement="top"
+          overlay={config.configured_at && (
+            <Tooltip variant="light">This integration has not been validated. Please follow the testing instructions to validate your integration.</Tooltip>
+          )}
+        >
+          <KeyOff
+            className="bg-danger-500 rounded-circle text-white mr-2 p-1"
+          />
+        </OverlayTrigger>
+      )}
+      {config.validated_at && !config.active && (
+        <KeyOff
+          className="bg-danger-500 rounded-circle text-white mr-2 p-1"
+        />
+      )}
+    </>
+  );
+
+  const renderCardBadge = () => (
+    <>
+      {(!config.validated_at || config.submitted_at > config.configured_at) && (
+        <Badge className="ml-4 p-2 font-weight-light" variant="light">In-progress</Badge>
+      )}
+      {config.validated_at && config.submitted_at < config.configured_at && !config.active && (
+        <Badge className="ml-4 p-2 font-weight-light" variant="light">Disabled</Badge>
+      )}
+    </>
+  );
+
+  const renderCardButton = () => (
+    <>
+      {!config.validated_at && (config.submitted_at < config.configured_at) && (
```

Comment:
nit: We're doing a lot of embedded state checks on the config in question.  I'm wondering if the logic would be easier to follow if we did a check up front to determine the state of the config (VALIDATED, DISABLED, ENABLED) and then base all the conditional renderings off that.